ghcide/src/Development/IDE/Spans/AtPoint.hs

```diff
@@ -225,10 +225,17 @@ atPoint IdeOptions{} (HAR _ hf _ _ kind) (DKMap dm km) env pos = listToMaybe $ p
         wrapHaskell x = "\n```haskell\n"<>x<>"\n```\n"
         info = nodeInfoH kind ast
         names = M.assocs $ nodeIdentifiers info
+        isInternal :: (Identifier, IdentifierDetails a) -> Bool
+        isInternal (Right n, _) =
+          let name = printOutputable n
+              prefix = T.take 2 name
+          in elem prefix ["$d", "$c"]
```

Comment:
Would you mind adding a comment, what we consider to be an internal name?

Reply:
Will do. I had done this locally right before pushing, but forgot to commit.

Reply:
Good thought, there is a list of GHC OccName prefixes in `Development.IDE.Plugin.Completions.Logic`.

Reply:
Re: tests, is there a good example test or test module I can use as a reference? It looks like there aren't existing tests for the `AtPoint` module. Should `ghcide` tests be placed within the top-level `test/` directory or the `ghcide/test/` directory?

I think we'll want to move the `isInternal` function into the `Development.IDE.Plugin.Completions.Logic` module, and perhaps give it a slightly better name. We could also extract the retrieval of names from the `atPoint` function, make it a top-level function (`getNamesAtPoint`?), and then test that directly rather than test the result of `atPoint`.

I'm new to HLS dev, so I'll proceed with whatever the experienced members recommend.

Reply:
Hey @drone-rites sorry for the late response. The ghcide test suite lives all in `ghcide/test/exe/Main.hs` and is almost 100% composed of integration tests, you will find almost no unit tests at all. The best place to test this change is:

https://github.com/haskell/haskell-language-server/blob/6dd5a3b29f857a73f2c526e93003b26928c3d71d/ghcide/test/exe/Main.hs#L996-L1090

Reply:
Hi @pepeiborra, thanks for the pointer; perfect timing as I was just refactoring the updates today.

Reply:
Added a single, minimal test that captures the case I was originally observing. Type class dictionary variables associated with a type definition were showing up in the hover info when cursor was placed on a comment above the type definition.